src/filter.rs

```diff
@@ -2,6 +2,185 @@ use core::convert::TryInto;
 
 use crate::{common::BytesPerPixel, Compression};
 
+#[cfg(feature = "unstable")]
+mod simd {
+    // Optional module that contains portable_simd versions of the most important unfiltering algorithms.
+    use core::convert::TryInto;
+    use core::simd::cmp::{SimdOrd, SimdPartialOrd};
+    use core::simd::num::{SimdInt, SimdUint};
+    use core::simd::{LaneCount, Simd, SupportedLaneCount};
+
+    // Import the fastest arch-specific scalar implementations from the outer crate.
+    #[cfg(target_arch = "aarch64")]
+    use crate::filter::filter_paeth as filter_paeth_decode;
+    #[cfg(target_arch = "x86_64")]
+    use crate::filter::filter_paeth_stbi as filter_paeth_decode;
```

Comment:
We should share this with the scalar logic (replacing the current `if` statement). And we'll want to pick something for non-x86_64 and non-aarch64 architectures to avoid a compile error.

Also, `filter_paeth_chosen` might be a good name?

Reply:
Fixed with new name in subsequent commit